pkg/recommender/metrics/mock_prometheus_api.go

```diff
@@ -0,0 +1,76 @@
+package metrics
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"time"
+)
+
+type MockApi struct{}
+
+var Q = func(query string, ts time.Time) (model.Value, v1.Warnings, error) {
```

Comment:
Why this isn't a field of MockApi?

Reply:
Changed it.
`type mockApi struct {
	q  func(query string, ts time.Time) (model.Value, v1.Warnings, error)
	qr func(query string, r v1.Range) (model.Value, v1.Warnings, error)
}`